File: Leetcode/485/485.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxConsec(0);
        int curConsec(0);
        int lastVal(-1);
        for (size_t i = 0; i != nums.size(); ++i) {
            if (nums[i] == 1 && lastVal != 1) {
                curConsec = 1;
                lastVal = 1;
            } else if (nums[i] == 1 && lastVal == 1) {
                ++curConsec;
            } else if (nums[i] != 1) {
                lastVal = nums[i];
                curConsec = 0;
            }
            maxConsec = max(maxConsec, curConsec);
        }
        return maxConsec;
    }
};

File: Leetcode/485/485.go
func findMaxConsecutiveOnes(nums []int) int {
	startIndex := -1
	ans := 0
	for endIndex := 0; endIndex < len(nums); endIndex++ {
		if nums[endIndex] == 1 {
			if startIndex == -1 {
				startIndex = endIndex
			}
		} else {
			if startIndex != -1 {
				if endIndex - startIndex > ans {
					ans = endIndex - startIndex
				}
				startIndex = -1
			}
		}
	}
	if startIndex != -1 && len(nums) - startIndex > ans {
		ans = len(nums) - startIndex
	}
	return ans
}